flyteplugins/go/tasks/plugins/k8s/spark/spark.go

```diff
@@ -412,9 +406,13 @@ func getEventInfoForSpark(pluginContext k8s.PluginContext, sj *sparkOp.SparkAppl
 			return nil, err
 		}
 
+		// "All user" logs are shown already in the queuing and initializing phase.
+		for _, log := range o.TaskLogs {
+			log.ShowWhilePending = true
+		}
+
 		taskLogs = append(taskLogs, o.TaskLogs...)
```

Comment:
If one takes a look at how the log links are currently configured for the spark plugin, one sees that "All User Logs" [are sent already when queued](https://github.com/flyteorg/flyte/blob/c899586448f842ae361b91ccb610d3d2c612546c/flyteplugins/go/tasks/plugins/k8s/spark/spark.go#L403) as the log plugin's init is not guarded by `if !isQueued {`.

This has the effect that the "All User" logs are shown already in the init phase (current master):

<img width="1415" alt="spark_master" src="https://github.com/flyteorg/flyte/assets/36511035/02677bbd-ba92-43fa-90a9-16dad56c234b">
(Fails because I don't have a proper spark image at hand.)


The log link is not shown in the queued phase currently even though the log link is added to `info` in the queued phase because the plugin doesn't currently bump the phase version when the reason changes meaning that only a single `task submitted to k8s` event is sent to admin in the queued phase. When bumping the phase version, the log link is shown already in the queued phase:

<img width="835" alt="spark_update_phase" src="https://github.com/flyteorg/flyte/assets/36511035/0befd8d5-1e45-4d53-b69a-9a398cce8c35">

Given `if !isQueued {` I feel this might have been the original intention. In practice this makes little difference as in the init phase, the driver pod is still pending, meaning that the transition from queued to initializing phase is almost instantaneous even if the actual scheduling of the pods takes longer.

With the removal of `if !isQueued {` and instead setting "All User" logs to `ShowWhilePending`, this is how the log links behave:

<img width="1424" alt="spark_branch" src="https://github.com/flyteorg/flyte/assets/36511035/eefd3b6d-dbd8-4cb4-b3f9-b69157f0fd8f">